DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs

```diff
@@ -25,49 +29,60 @@ public JobProfileIndexEnhancer(IJobProfileRepository jobProfileRepository, IJobP
+            this.jobProfileIndex = jobProfileIndex;
         }
 
-        public JobProfileIndex GetRelatedFieldsWithUrl(JobProfileIndex jobProfileIndex)
+        public void PopulateRelatedFieldsWithUrl()
         {
             if (JobProfile != null)
             {
-                jobProfileIndex.JobProfileCategoriesWithUrl = GetJobProfileCategoriesWithUrl(JobProfile);
+                jobProfileIndex.JobProfileCategoriesWithUrl = GetJobProfileCategoriesWithUrl();
 
-                jobProfileIndex.Interests = JobProfile.RelatedInterests;
-                jobProfileIndex.Enablers = JobProfile.RelatedEnablers;
-                jobProfileIndex.EntryQualifications = JobProfile.RelatedEntryQualifications;
-                jobProfileIndex.TrainingRoutes = JobProfile.RelatedTrainingRoutes;
-                jobProfileIndex.JobAreas = JobProfile.RelatedJobAreas;
-                jobProfileIndex.PreferredTaskTypes = JobProfile.RelatedPreferredTaskTypes;
+                jobProfileIndex.Interests = JobProfile.RelatedInterests.ToList();
+                jobProfileIndex.Enablers = JobProfile.RelatedEnablers.ToList();
+                jobProfileIndex.EntryQualifications = JobProfile.RelatedEntryQualifications.ToList();
+                jobProfileIndex.TrainingRoutes = JobProfile.RelatedTrainingRoutes.ToList();
+                jobProfileIndex.JobAreas = JobProfile.RelatedJobAreas.ToList();
+                jobProfileIndex.PreferredTaskTypes = JobProfile.RelatedPreferredTaskTypes.ToList();
             }
-
-            return jobProfileIndex;
         }
 
-        public async Task<JobProfileIndex> GetSalaryRangeAsync(JobProfileIndex jobProfileIndex)
+        public Task PopulateSalary()
         {
             // Conversions taking place because sitefinity returns Decimal and Azure Search accepts Double fields
-            if (JobProfile.IsLMISalaryFeedOverriden.HasValue && JobProfile.IsLMISalaryFeedOverriden.Value != true && !string.IsNullOrWhiteSpace(JobProfile.SOCCode))
+            if (JobProfile.IsLMISalaryFeedOverriden == true)
             {
-                var salary = await salaryService.GetSalaryBySocAsync(JobProfile.SOCCode);
-                jobProfileIndex.SalaryStarter = Convert.ToDouble(salaryCalculator.GetStarterSalary(salary));
-                jobProfileIndex.SalaryExperienced = Convert.ToDouble(salaryCalculator.GetExperiencedSalary(salary));
+                jobProfileIndex.SalaryStarter = Convert.ToDouble(JobProfile.SalaryStarter);
+                jobProfileIndex.SalaryExperienced = Convert.ToDouble(JobProfile.SalaryExperienced);
             }
             else
             {
-                jobProfileIndex.SalaryStarter = Convert.ToDouble(JobProfile.SalaryStarter);
-                jobProfileIndex.SalaryExperienced = Convert.ToDouble(JobProfile.SalaryExperienced);
+                //Mutate soc code
+                var socCode = JobProfile.SOCCode;
+                return Task.Run(() => PopulateSalaryrFromLMIAsync(socCode));
```

Comment:
on Dev testing there were cases were this was not populated and this would cause issues down stream, might be worth putting a check here before calling the API